sydent/http/servlets/replication.py

```diff
@@ -16,22 +16,51 @@
 
 import twisted.python.log
 from twisted.web.resource import Resource
+from twisted.web import server
+from twisted.internet import defer
 from sydent.http.servlets import jsonwrap
 from sydent.threepid import threePidAssocFromDict
 from sydent.db.peers import PeerStore
 from sydent.db.threepid_associations import GlobalAssociationStore
+from sydent.db.invite_tokens import JoinTokenStore
+from sydent.replication.peer import NoMatchingSignatureException, NoSignaturesException, RemotePeerError
+from signedjson.sign import SignatureVerifyException
 
 import logging
 import json
 
 logger = logging.getLogger(__name__)
 
+MAX_SG_ASSOCS_LIMIT = 100
+MAX_INVITE_TOKENS_LIMIT = 100
+MAX_EPHEMERAL_PUBLIC_KEYS_LIMIT = 100
+
 class ReplicationPushServlet(Resource):
     def __init__(self, sydent):
         self.sydent = sydent
 
-    @jsonwrap
     def render_POST(self, request):
+        self._async_render_POST(request)
+        return server.NOT_DONE_YET
+
+    @defer.inlineCallbacks
+    def _async_render_POST(self, request):
```

Comment:
I have a general concern with the structure of this method that there are a whole bunch of things that might throw exceptions (eg, updating the database) which aren't caught, and when that happens the request will be left open for the caller to time out.

Ideally the whole `_async_render_POST` method would be wrapped in a `try/except` which returned a 500 in case of an uncaught exception.

However, I think that is a more general issue in sydent as a whole, so probably one to fix in a separate PR. Something to bear in mind if you're looking at robustness and that sort of thing.